src/Shared/UnitTests/Vacancies.Client/Infrastructure/Services/TrainingProviderServiceTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Application.Configuration;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Services.TrainingProvider;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Moq;
+using SFA.DAS.Apprenticeships.Api.Types.Providers;
+using SFA.DAS.Providers.Api.Client;
+using Xunit;
+
+namespace Esfa.Recruit.Vacancies.Client.UnitTests.Vacancies.Client.Infrastructure.Services
+{
+    public class TrainingProviderServiceTests
+    {
+        [Fact]
+        public async Task GetProviderAsync_ShouldReturnEsfaTestTrainingProvider()
+        {
+            var loggerMock = new Mock<ILogger<TrainingProviderService>>();
+            var providerApiMock = new Mock<IProviderApiClient>();
+
+            var sut = new TrainingProviderService(loggerMock.Object, providerApiMock.Object);
+
+            var provider = await sut.GetProviderAsync(EsfaTestTrainingProvider.Ukprn);
+
+            provider.Ukprn.Should().Be(EsfaTestTrainingProvider.Ukprn);
+            provider.Name.Should().Be(EsfaTestTrainingProvider.Name);
+            provider.Address.Postcode.Should().Be("CV1 2WT");
+
+            providerApiMock.Verify(p => p.GetAsync(It.IsAny<long>()), Times.Never);
+        }
+
+        [Fact]
+        public async Task GetProviderAsync_ShouldAttemptToFindTrainingProvider()
+        {
+            const long ukprn = 88888888;
+
+            var loggerMock = new Mock<ILogger<TrainingProviderService>>();
+            var providerApiMock = new Mock<IProviderApiClient>();
+            providerApiMock.Setup(p => p.GetAsync(ukprn))
+                .ReturnsAsync(new Provider
+                {
+                    Ukprn = ukprn,
+                    ProviderName = "name",
+                    Addresses = new List<ContactAddress>
+                    {
+                        new ContactAddress
+                        {
+                            ContactType = "something else",
+                            Primary = "ignored",
+                            Secondary = "ignored",
+                            Street = "ignored",
+                            Town = "ignored",
+                            PostCode = "ignored"
+                        },
+                        new ContactAddress
```

Comment:
If you hold this in a variable, then you can use this in assertion instead of repeating the strings.